src/main/kotlin/org/rust/ide/intentions/DemorgansLawIntention.kt

```diff
@@ -0,0 +1,107 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.tree.IElementType
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.elementType
+import org.rust.lang.core.psi.util.parentOfType
+
+class DemorgansLawIntention : PsiElementBaseIntentionAction() {
+    override fun getText() = "DeMorgan Law"
```

Comment:
I would introduce base class [MutablyNamedIntention](https://upsource.jetbrains.com/idea-ce/file/idea-ce-cfc493fae3127d133589d446cc535d8ac51b72c5/plugins/IntentionPowerPak/src/com/siyeh/ipp/base/MutablyNamedIntention.java?nav=stub:6:focused) and implement intention's name like [here](https://upsource.jetbrains.com/idea-ce/file/idea-ce-cfc493fae3127d133589d446cc535d8ac51b72c5/plugins/IntentionPowerPak/src/com/siyeh/ipp/bool/DemorgansIntention.java?nav=1148:1607:focused&line=0), so we would get nice `Replace '&&' with '||'` and vice versa names.

Of course, family name should stay as it is (sans typos): `De Morgan's laws`